### DISCUSSION TITLE
How to avoid the creation of a directory structure if no backup is found?

### DISCUSSION THREAD

**Discussion by rosnu** [borgbackup/borg#7887] [Q&A] [answered]
Hi,
My Borg backup normally goes to a USB drive, but it is not always available. In this case, Borg of course aborts with an error, but then creates some empty directories (its basic structure), with the consequence that the USB drive is mounted to a different path on the next mount (the old path is indexed) and Borg can no longer find it. Is there a simple solution for this (avoid creating this empty directories), or do I have to do some tinkering?

**Comment by ThomasWaldmann** [comment]
It's unclear what you mean, please show it.

**Reply by rosnu** [reply]
Ok, sorry, I should probably first check if Nextcloud, which uses Borg, might create the path itself, and not Borg itself.

**Reply by rosnu** [reply, marked as answer]
The root cause lies in Docker's default handling of volumes and mounting. Docker creates an empty directory if the host directory to be mounted is not available. This behavior also applies to Nextcloud-Borg volumes. This is not directly related to Nextcloud or Borg; it's more about the configuration.

I will likely create a script that, before running Borg Backup, mounts the correct backup paths if necessary.
